sage-ios/SAGE/SAGE/ProfileImageView.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  ProfileImageView.swift
+//  SAGE
+//
+//  Created by Andrew Millman on 2/5/16.
+//  Copyright © 2016 Cal Blueprint. All rights reserved.
+//
+
+import UIKit
+
+class ProfileImageView: UIView {
+    
+    private var imageView = UIImageView()
+    private var badgeBorder = UIView()
+    private var badgeView = UILabel()
+    
+    private let badgeSizeFactor: CGFloat = 1.80
+    private let badgeViewPercentage: CGFloat = 0.90
+    
+    //
+    // MARK: - Initialization
+    //
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.setupSubviews()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    private func setupSubviews() {
+        self.imageView.contentMode = .ScaleAspectFill
+        self.imageView.clipsToBounds = true
+        self.addSubview(self.imageView)
+        
+        self.badgeBorder.backgroundColor = UIColor.whiteColor()
+        self.badgeBorder.alpha = 0
+        self.addSubview(self.badgeBorder)
+        
+        self.badgeView.textColor = UIColor.whiteColor()
+        self.badgeView.font = UIFont.getBoldFont()
+        self.badgeView.textAlignment = .Center
+        self.badgeView.clipsToBounds = true
+        self.badgeBorder.addSubview(self.badgeView)
+    }
+    
+    //
+    // MARK: - Layout
+    //
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        let radius =  CGRectGetWidth(self.frame) / 2
+        let xTranslation = radius/sqrt(2)
+        let yTranslation = radius/sqrt(2)
+        
+        self.imageView.setX(0)
+        self.imageView.setY(0)
+        
+        self.badgeBorder.centerInSuperview()
+        self.badgeBorder.moveX(xTranslation)
+        self.badgeBorder.moveY(yTranslation)
+        self.badgeBorder.frame = CGRectIntegral(self.badgeBorder.frame)
+        self.badgeBorder.layer.cornerRadius = CGRectGetWidth(self.badgeBorder.frame)/2
+        
+        self.badgeView.centerInSuperview()
+    }
+    
+    //
+    // MARK: - Public Methods
+    //
+    func setDiameter(diameter: CGFloat) {
+        self.setSize(width: diameter, height: diameter)
+        self.imageView.setSize(width: diameter, height: diameter)
+        self.imageView.layer.cornerRadius = diameter / 2
```

Comment:
to be consistent, all the layer should either be in this function or in layout subviews